lib/ramble/ramble/language/application_language.py

```diff
@@ -71,13 +71,22 @@ def workload(name, executables=None, executable=None, input=None,
     """
 
     def _execute_workload(app):
+        exp_executables = executables
+        if isinstance(exp_executables, list):
+            exp_executables = ramble.language.language_helpers.expand_patterns(app.executables,
```

Comment:
Could this logic be pushed down into `require_definition` or `merge_definition`? (i.e. the list check that then expands and merges with the singular definition)

Reply:
Definitely. I've updated this to be in `merge_definition` and it's much cleaner now.